### PR TITLE
Migrate `PreviewTransactionDto` to `zod`

### PR DESCRIPTION
**Pull request by iamacook**
## Summary

This migrates the validation of `PreviewTransactionDto` to `zod`.

## Changes

- Remove `PreviewTransactionDtoValidationPipe`
- Add and use `PreviewTransactionDtoSchema` schema
- Infer `PreviewTransactionDto` from schema
- Update tests accordingly

### CODE REVIEW

File: src/routes/transactions/__tests__/controllers/preview-transaction.transactions.controller.spec.ts [hunk 1]
```diff
@@ -38,18 +37,17 @@ describe('Preview transaction - Transactions Controller (Unit)', () => {
...
+    })
+      .overrideModule(AccountDataSourceModule)
+      .useModule(TestAccountDataSourceModule)
+      .overrideModule(CacheModule)
+      .useModule(TestCacheModule)
+      .overrideModule(RequestScopedLoggingModule)
+      .useModule(TestLoggingModule)
+      .overrideModule(NetworkModule)
+      .useModule(TestNetworkModule)
+      .compile();
```
Comment (iamacook) [inline src/routes/transactions/__tests__/controllers/preview-transaction.transactions.controller.spec.ts:50]: This change was required in order to test validation.

**Comment by iamacook** [comment]
Marking as a draft until the dependant is finalised.

File: src/domain/safe/entities/operation.entity.ts [hunk 2]
```diff
@@ -1,6 +1,4 @@
-export const CALL_OPERATION = 0;
-export const DELEGATE_OPERATION = 1;
-
-export type Call = typeof CALL_OPERATION;
-export type Delegate = typeof DELEGATE_OPERATION;
-export type Operation = Call | Delegate;
+export enum Operation {
+  CALL = 0,
+  DELEGATE = 1,
+}
```
Comment (fmrsabino) [inline src/domain/safe/entities/operation.entity.ts:4]: This is used by more entities than the `PreviewTransactionDto`. I would isolate this change first.

**Review by fmrsabino — CHANGES_REQUESTED** [review]

File: src/domain/safe/entities/operation.entity.ts (hunk 2 above)
Comment (iamacook) [reply src/domain/safe/entities/operation.entity.ts]: I reverted the simplification in 1ebbadfa4a0de17ec5417785c91460b244969784.

**Review by fmrsabino — APPROVED** [review]

**Pull request merged**